src/Microsoft.Graph.Auth/Extensions/IClientApplicationBaseExtensions.cs

```diff
@@ -61,5 +64,24 @@ internal static async Task<AuthenticationResult> GetAccessTokenSilentAsync(this
                         exception);
             }
         }
+
+        /// <summary>
+        /// Determines if an authority url has a `well known` tenant name (common, consumers, organizations) in its tenant segment.
+        /// </summary>
+        /// <param name="currentAuthority">The authority url to check.</param>
+        /// <returns>True is pre-defined tenant names are present, and false if not. </returns>
+        internal static bool ContainsWellKnownTenantName(string currentAuthority)
+        {
+            if (!Uri.IsWellFormedUriString(currentAuthority, UriKind.Absolute))
+                throw new ArgumentException("Invalid authority URI set.");
+
+            Uri authorityUri = new Uri(currentAuthority);
+
+            return (new List<string>
```

Comment:
As this gets called on every call and that list is constant, I suggest we make it static rather than re-creating it every time.